src/GraphODataTemplateWriter/Extensions/OdcmModelExtensions.cs

```diff
@@ -525,5 +525,42 @@ public static IEnumerable<OdcmMethod> MethodsAndOverloads(this OdcmClass odcmCla
         {
             return odcmClass.Methods.SelectMany(x => x.WithOverloads());
         }
+
+        /// <summary>
+        /// Use this method to get a collection of navigation properties on the return type 
+        /// of a composable function. 
+        /// </summary>
+        /// <param name="odcmMethod">The OdcmMethod to target.</param>
+        /// <returns>An ordered (by name) list of navigation properties bound 
+        /// to the return type. Can be an empty list.</returns>
+        public static List<OdcmProperty> GetComposableFunctionReturnTypeNavigations(this OdcmMethod odcmMethod)
+        {
+            if (!odcmMethod.IsComposable)
+                throw new InvalidOperationException("This extension method is intended " +
+                                                    "to only be called on a composable function.");
+
+            return (odcmMethod.ReturnType as OdcmClass).Properties
```

Comment:
the cast might return null, null propagation?

Reply:
Same response as the NRE above. The fatal exception triggers us to investigate issues so that we can resume generation.